### PR TITLE
feat(pipeline): add destination logs

### PR DESCRIPTION
**Pull request by thmsobrmlr**
## Problem

We need logs for batch exports and regular apps.

## Changes

This PR adds them.

## How did you test this code?

Added logs locally + a storybook story

```sql
INSERT INTO log_entries (team_id, log_source, log_source_id, message, timestamp, level)
VALUES (1, 'batch_exports', '018cf3ad-af56-0001-68cc-3c03ee26c734', 'my log', NOW(), 'INFO');

SELECT * FROM log_entries WHERE log_source = 'batch_exports';
```

### CODE REVIEW

**Comment by Twixes** [comment]
👀 Can this be marked ready for review or needs a moment longer? @thmsobrmlr

**Comment by thmsobrmlr** [comment]
> 👀 Can this be marked ready for review or needs a moment longer? @thmsobrmlr

Wanted to wait for the CI run before I mark it as ready. But it's ready for review in general. Did you get a notification already?

**Comment by Twixes** [comment]
Yeah, GitHub sends a notification ASAP apparently 😄

File: frontend/src/scenes/pipeline/destinationsLogic.tsx
Comment (tiina303) [inline frontend/src/scenes/pipeline/destinationsLogic.tsx:43]: lets move this to `utils.tsx`
```suggestion
export enum PipelineAppType {
    BatchExport = 'batch_export',
    Plugin = 'plugin',
}
```
Because logs are broader - destination plugins logs are the same as transformation plugins logs.

Let's use the convention:
- 'kind' - filter/transformation/destination
- 'type' - batch_export or plugin

Comment (Twixes) [reply frontend/src/scenes/pipeline/destinationsLogic.tsx:43]: We found that the same applies in https://github.com/PostHog/posthog/pull/19702#discussion_r1448718181, so I'll actually do this refactor in a new PR as soon as this PR's merged

File: frontend/src/types.ts
Comment (tiina303) [inline frontend/src/types.ts:1635]: could we move this one also into `scenes/pipeline/pipelineAppLogsLogic`

File: frontend/src/scenes/pipeline/pipelineAppLogsLogic.tsx
Comment (tiina303) [inline frontend/src/scenes/pipeline/pipelineAppLogsLogic.tsx:28]:
```suggestion
    type: PipelineAppType
```

File: frontend/src/scenes/pipeline/pipelineAppLogsLogic.tsx
Comment (tiina303) [inline frontend/src/scenes/pipeline/pipelineAppLogsLogic.tsx:207]: maybe would be nice if we did it the same way as with DestinationType - we could have PipelineAppLogType - then a selector, where we combine it from the two different options, which if there's a third thing latter and or any changes further down you'll only need to operate on a single type would be simpler 🤷

**Review by Twixes — APPROVED** [review]

**Pull request merged**
